account_analytic_parent/models/account_analytic_account.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Luxim d.o.o.
+# Copyright 2017 Matmoz d.o.o.
+# Copyright 2017 Deneroteam.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = "account.analytic.account"
+
+    parent_id = fields.Many2one(
+        comodel_name="account.analytic.account",
+        string="Parent Analytic Account"
+    )
+    child_ids = fields.One2many(
+        comodel_name="account.analytic.account",
+        inverse_name="parent_id",
+        string="Child Accounts", copy=True)
+
+    @api.multi
+    def _compute_debit_credit_balance(self):
+        """
+        Warning, this method overwrites the standard because the hierarchy
+        of analytic account changes
+        """
+        super(AccountAnalyticAccount, self)._compute_debit_credit_balance()
+        for account in self:
+            account.debit += sum(account.mapped("child_ids.debit"))
```

Comment:
For these 3 lines, I think it should be a `=` instead of a `+=`.

`account.debit = sum(account.mapped("child_ids.debit"))`

Reply:
I think the code is ok as it is. The call to super computes debit, credit and balance for every account. Then the method adds debit, credit and balance from child accounts to parent. If we use `=` instead of `+=`, we lose the information of the account itself and only get the sum from its children.

Reply:
I think the code is ok as it is too.